src/doom/am_map.c

```diff
@@ -1545,20 +1545,21 @@ void AM_drawWalls(void)
 	    {
 		// [crispy] draw keyed doors in their respective colors
 		// (no Boom multiple keys)
+		// make keyed doors flash for easier visibility
 		keycolor_t amd;
 		if (!(lines[i].flags & ML_SECRET) &&
 		    (amd = AM_DoorColor(lines[i].special)) > no_key)
 		{
 		    switch (amd)
 		    {
 			case blue_key:
-			    AM_drawMline(&l, BLUES);
+			    AM_drawMline(&l, ((gametic % 35 < 18) ? BLUES : BLACK));
```

Comment:
It doesn't have to be exactly 0.5 sec. Does `(leveltime & 15)` work just as well for you?

Reply:
`(leveltime & 16)` does. `15` only flashes for one tic.